### PR TITLE
Allow searching single file history with prefix "/" 

### PR DESCRIPTION
**Pull request by soleo**
## What is this PR trying to solve

Refer to #76 

This PR enables a user to search a single file's commit history. The equivalent git command would be `git log --follow path/to/filename`. The difference between the git command and a gitUp single file search is we're using prefix `/` to indicate that we only want the file history.

e.g.

If you want to search any changes for file `package.json`, you'd use `/package.json` in search.

If you want to search for the keyword `package.json` in commit messages, you'd use `package.json` instead.

## What it looks like
[
screen shot 2018-02-25 at 1 54 22 pm
](https://user-images.githubusercontent.com/635858/36645962-51ec8674-1a36-11e8-95a1-50a6f0c2e7b6.png)

## Known issue

When searching in a large repo, it is slow

## Contribution Agreement

I AGREE TO THE GITUP CONTRIBUTOR LICENSE AGREEMENT

### CODE REVIEW

File: GitUpKit/Core/GCLiveRepository.m [hunk 1]
```diff
@@ -1008,8 +1008,18 @@ static BOOL _MatchReference(NSString* match, NSString* name) {
 - (NSArray*)findCommitsMatching:(NSString*)match {
   XLOG_DEBUG_CHECK(_database);
   NSMutableArray* results = [[NSMutableArray alloc] init];
+  
   match = [match stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-  if (match.length >= kMinSearchLength) {
+  bool searchFileHistoryOnly = [match hasPrefix:@"/"];
+  // Search file history directly
+  if(match.length >= kMinSearchLength && searchFileHistoryOnly) {
```
Comment (swisspol) [inline GitUpKit/Core/GCLiveRepository.m]: This needs to be (kMinSearchLength + 1) because of prefix /

Comment (soleo) [reply GitUpKit/Core/GCLiveRepository.m]: Will change it in next commit.

**Comment by soleo** [comment]
@swisspol Fix the issues you mentioned in the comments. However, I cannot lint the code with `format-source.sh` because homebrew now uses clang-format v7.0.0 instead of v6.0.0. Should we update the version and push another commit with v7.0.0-linted code or shall we use do another PR to fix the version?

**Comment by swisspol** [comment]
Try rebasing on master and running `format-source.sh` again (only pick changes on your files though).

**Pull request closed without merging**